cms_coding_rules.md

```diff
@@ -88,10 +88,15 @@ If necessary to create a unique name, one can add the directory name:
 27. Provide meaningful argument names in method declarations in the header ﬁle to indicate usage, unless the type fully describes the usage.
 28. Try to avoid excessively long lines of code that impair readability.
 29. Data members of a class must not be redeﬁned in derived classes since doing so hides the original data member and could create confusion between the original and redefined data members.
+30. Use `std::abs()` instead of `fabs()`.
```

Comment:
Why only `abs` and not all the math functions ?

Reply:
yes why `abs` only? By the way, clang-tidy has this check https://clang.llvm.org/extra/clang-tidy/checks/performance-type-promotion-in-math-fn.html but this will only apply to math functions with implicit `float` to `double` conversion.

Reply:
https://blog.audio-tk.com/2018/03/20/writing-custom-checks-for-clang-tidy/ suggests how to add clang-tidy checks for such deprecated functions. @gartung may be we can include these in our llvm build?

Reply:
Yes a custom checker can be made for that.

Reply:
@gartung , have to looked in to this? Can we add a custom checker for this?

Reply:
No, but I made a checker for narrowing cast. It should be straight-forward to change it to check the reverse\.

Reply:
I meant  a static analyzer checker.

Reply:
sorry to jump in now in this old thread, but I see developers using `fabs` instead of `std::abs` (and even worse using `abs` for floats) . Was the ckecker discussed above made or is there any objection in moving on with adding this 4.30 rule?

Reply:
I'm not so sure any more that we should discourage the use of `fabs`.
In fact, people may end up using a lot more conversions to/from doubles without realising it:
```c++
auto x = f(...);            // returns a float
auto y = std::abs(x * 2.);  // returns a double
```

At least `fabs` _etc._ are explicit about being single precision.

Reply:
> I'm not so sure any more that we should discourage the use of fabs.

good observation, will keep in mind while reviewing.

Reply:
the case about `x * 2.` should probably be formalized as a rule/reminder that single precision code should rely on single precision literals.
`auto y = std::fabs(x * 2.)` where x is a float should still return a double; `std::fabsf` would be a float with a warning if a `2.` is used.

Reply:
Sorry, yes, `fabfs`.